docs/product-user-guide/src/main/asciidoc/drl-rules-central-create-proc.adoc

```diff
@@ -0,0 +1,182 @@
++
+.Query and query results for people under the age of 21, with a rule
+[source,java]
+----
+query "people over the age of 30"
+    person : Person( age < 21 )
+end
+
+QueryResults results = ksession.getQueryResults( "people under the age of 21" );
+System.out.println( "we have " + results.size() + " people under the age  of 21" );
+
+System.out.println( "These people are are under 21:" );
+}
+
+rule "Underage"
+  when
+    application : LoanApplication( )
+    Applicant( age < 21 )
+  then
+    application.setApproved( false );
+    application.setExplanation( "Underage" );
+end
+----
++
+* `declare`: (optional) Use this to declare a new fact type to be used by rules in the DRL file. The default fact type in DRL files is `Object`, but you can declare other types as needed. Declaring fact types in DRL files enables you to define a new fact model directly in the decision engine, without creating models in a lower-level language like Java. *<@Michael (6 of 13): This right? From my deductions with all the info out there.>*
++
+.Declaring and using a new fact type
+[source,java]
+----
+declare Person
+  name : String
+  dateOfBirth : java.util.Date
+  address : Address
+end
+
+rule "Using a declared type"
+  when
+    $p : Person( name == "James" )
+  then   // Insert Mark, who is a customer of James.
+    Person mark = new Person();
+    mark.setName("Mark");
+    insert( mark );
+end
+----
++
+* `rule`: Use this to define each rule in the DRL file. Rules consist of a rule name in the format `rule "name"`, followed by optional attributes that define rule behavior (such as `salience` or `no-loop`), followed by `when` and `then` definitions. The same rule name cannot be used more than once in the same package. The `when` part of the rule contains the conditions that must be met to execute an action. For example, if a bank requires loan applicants to have over 21 years of age, then the `when` condition for an "Underage" rule would be `Applicant( age < 21 )`. The `then` part of the rule contains the actions to be performed when the conditional part of the rule has been met. For example, when the loan applicant is under 21 years old, the `then` action would be `setApproved( false )`, declining the loan because the applicant is under age. Conditions (`when`) and actions (`then`) consist of a series of stated fact patterns with optional constraints, bindings, and other supported Java elements, based on the available data objects in the package. These patterns determine how defined objects are affected by the rule.
++
+.Rule for loan application age limit
+[source,java]
+----
+rule "Underage"
+  salience 15
+  dialect "mvel"
+  when
+    application : LoanApplication( )
```

Comment:
@sterobin You need to be careful with these types of example. There is not implicit relation between one instance of ```LoanApplication``` and one instance of ```Applicant``` so this rule would trigger for all combinations of ```LoanApplication``` instances and all ```Applicant``` instances; i.e. a cross-product. 

The "mortgages" object model is pretty poor; ideally a ```LoanApplication``` would apply to one ```Applicant``` in which case the following rule would ensure the correct ```LoanApplication``` would relate to the correct ```Applicant```.
```
rule "Underage"
when
a : Applicant( age < 21 )
la : LoanApplication( applicant == a )
then
la.setApproved( false );
end
```
The "mortgages" data model does not have an instance field ```applicant``` in ```LoanApplication```.

Reply:
@manstis , I'm not sure what to do here because this example is literally copied over from the source extraction of the example guided rule for this as it appears in Central. I didn't just come up with it. It's defined as part of the example loaded with the product. I figured if it's an example there, it's safe to go with. But it sounds like you're saying the mortgages sample project in Central in general is poor. Per Jozef's recommendation, I've tried to stick with the same sample project throughout the docs (except in rule templates, for reasons) for consistency and for more concrete examples, and then rely on your and him to correct me where I'm way off.

Point being, I can definitely make this rule here better, as you've described, but that the sample is a poor one in Central, I can't do much about I guess. Especially now that I've used it throughout all my docs.

Do you have another sample project in Central that you recommend instead, for future reference?

Reply:
@sterobin Go with what you have... the example is clear it its intent.